subprojects/launcher/src/integTest/groovy/org/gradle/launcher/continuous/SmokeContinuousIntegrationTest.groovy

```diff
@@ -332,10 +452,14 @@ class SmokeContinuousIntegrationTest extends AbstractContinuousIntegrationTest {
 
     def "ignores non source when source is empty"() {
         when:
+        file("source").createDir()
```

Comment:
Why change this? Shouldn't we add a separate test for when `source` exists instead?

Reply:
This is the test when the source is empty, aka an empty directory. The test for no existing files is `detects changes when no files are in the project`.